install/compilation_options.md

```diff
@@ -0,0 +1,247 @@
+cd $SCRIPT_DIR
+if [ ! -f "${LIBARCHIVE_ARCHIVE}" ]; then
+  if [ -x "$(command -v wget)" ]; then
+    wget ${LIBARCHIVE_DOWNLOAD_URL}/${LIBARCHIVE_ARCHIVE} || { echo "Error while downloading Libarchive."; exit 1; }
+  else
+    echo "You need to install wget."
+    exit 1
+  fi
+fi
+if [ ! -d "$SCRIPT_DIR/$LIBARCHIVE_DIRECTORY" ]; then
+  tar -xzf $LIBARCHIVE_ARCHIVE -C $SCRIPT_DIR || { echo "Error while extracting Libarchive."; exit 1; }
+fi
+if [ ! -d "$BUILD_DIR" ]; then
+  mkdir -p $BUILD_DIR
+fi
+if [ ! -d "$INSTALL_DIR" ]; then
+  mkdir -p $INSTALL_DIR
+fi
+cd $BUILD_DIR
+cmake "$SCRIPT_DIR/$LIBARCHIVE_DIRECTORY" -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DCMAKE_C_COMPILER=$C_COMPILER || { echo "Error while cmake configuration of Libarchive."; exit 1; }
+make -j $NB_PROCESSORS_USED || { echo "Error while make of Libarchive."; exit 1; }
+make install || { echo "Error while make install of Libarchive."; exit 1; }' > compile_libarchive.sh
+$> chmod +x compile_libarchive.sh
+$> ./compile_libarchive.sh
+```
+
+## Boost
+
+[Boost website](https://www.boost.org/)
+
+``` bash
+$> echo 'BOOST_VERSION=1_69_0
+BOOST_ARCHIVE=boost_${BOOST_VERSION}.tar.gz
+BOOST_DIRECTORY=boost_$BOOST_VERSION
+BOOST_DOWNLOAD_URL=https://sourceforge.net/projects/boost/files/boost/${BOOST_VERSION//_/.}
+SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+BUILD_DIR=$SCRIPT_DIR/boost/build/${BOOST_VERSION//_/.}
+BUILD_TYPE=Release
+INSTALL_DIR=$SCRIPT_DIR/boost/install/${BOOST_VERSION//_/.}
+C_COMPILER=$(command -v gcc)
+CXX11_ENABLED=NO
+NB_PROCESSORS_USED=1
+cd $SCRIPT_DIR
+if [ ! -f "${BOOST_ARCHIVE}" ]; then
+  if [ -x "$(command -v wget)" ]; then
+    wget ${BOOST_DOWNLOAD_URL}/${BOOST_ARCHIVE} || { echo "Error while downloading Boost."; exit 1; }
+  else
+    echo "You need to install wget."
+    exit 1
+  fi
+fi
+if [ ! -d "$SCRIPT_DIR/$BOOST_DIRECTORY" ]; then
+  tar -xzf $BOOST_ARCHIVE -C $SCRIPT_DIR || { echo "Error while extracting Boost."; exit 1; }
+fi
+if [ "$(echo "$CXX11_ENABLED" | tr "[:upper:]" "[:lower:]")" = "no" -o "$(echo "$CXX11_ENABLED" | tr "[:upper:]" "[:lower:]")" = "false" -o "$(echo "$CXX11_ENABLED" | tr "[:upper:]" "[:lower:]")" = "off" ]; then
+  CXX_STD_FLAG="98"
+else
+  CXX_STD_FLAG="11"
+fi
+if [ ! -d "$BUILD_DIR" ]; then
+  mkdir -p $BUILD_DIR
+fi
+if [ ! -d "$INSTALL_DIR" ]; then
+  mkdir -p $INSTALL_DIR
+fi
+cd $SCRIPT_DIR/$BOOST_DIRECTORY
+./bootstrap.sh --prefix=$INSTALL_DIR cxxstd=$CXX_STD_FLAG --with-toolset=$(basename $C_COMPILER) --with-libraries=filesystem,program_options,serialization,system,log,iostreams || { echo "Error while bootstrap."; exit 1; }
+./b2 -d2 -j $NB_PROCESSORS_USED --build-dir=$BUILD_DIR  cxxflags="-std=c++$CXX_STD_FLAG" toolset=$(basename $C_COMPILER) variant=${BUILD_TYPE,,} install || { echo "Error while b2."; exit 1; }' > compile_boost.sh
+$> chmod +x compile_boost.sh
+$> ./compile_boost.sh
+```
+
+Version 1_69_0 was not tested on Centos6.4 but 1_59_0 is working well.
```

Comment:
Perhaps bold font or warning as for googletest?

Reply:
yes